xilem/examples/elm.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use masonry::widget::{CrossAxisAlignment, MainAxisAlignment};
+use winit::error::EventLoopError;
+use xilem::{
+    view::{button, flex, label},
+    EventLoop, WidgetView, Xilem,
+};
+use xilem_core::{adapt, map_action, map_state, MessageResult};
+
+#[derive(Default)]
+struct AppState {
+    adapt_count: i32,
+    map_state_count: i32,
+    map_action_count: i32,
+}
+
+// `map_state()` maps a subset of the state from the parent, such that views can be modularized by state
+fn map_state_view(count: &mut i32) -> impl WidgetView<i32> {
+    flex((
+        label(format!("adapt state count: {}", count)),
+        button("+", |count| *count += 1),
+        button("-", |count| *count -= 1),
+    ))
+}
+
+enum CountMessage {
+    Increment,
+    Decrement,
+}
+
+// `map_action()` is basically how elm works, i.e. provide a message that the parent view has to handle to update the state.
+// In this case the parent adjusts the count that is given to this view according to the message
+fn map_action_view<T>(count: i32) -> impl WidgetView<T, CountMessage> {
+    flex((
+        label(format!("adapt state count: {}", count)),
```

Comment:
This string value is a bit weird. Also, is there a reason this doesn't use the `{count}` shorthand?

Reply:
Whoops, you're right, probably an artifact resulted from copy/paste. I changed all the view names and their label (elm/direct/adapt counter). No there's no reason, I changed it.